knime_extension/src/nodes/spatialtool.py

```diff
@@ -299,9 +308,21 @@ def execute(self, exec_context: knext.ExecutionContext, left_input, right_input)
         )
         knut.check_canceled(exec_context)
         right_gdf.to_crs(left_gdf.crs, inplace=True)
-        gdf = left_gdf.sjoin(
-            right_gdf, how=self.join_mode.lower(), predicate=self.match_mode.lower()
-        )
+        if self.match_mode.lower() not in ["have_their_center_in", "contain_center_of"]:
+            gdf = left_gdf.sjoin(
+                right_gdf, how=self.join_mode.lower(), predicate=self.match_mode.lower()
+            )
+        elif self.match_mode.lower() == "have_their_center_in":
+            left_gdf["rep_center"] = left_gdf.representative_point()
```

Comment:
Make sure that this column name does not exist

Reply:
Use   get_unique_column_name to get the name

elif self.match_mode.lower() == "have_their_center_in":
            REP_Center= knut.get_unique_column_name("rep_center", left_input.schema) 
            left_gdf[REP_Center] = left_gdf.representative_point()
            left_gdf_temp = left_gdf.set_geometry(REP_Center)
            gdf = gp.sjoin(left_gdf_temp, right_gdf, predicate="within")
            gdf = gdf.set_geometry(self.left_geo_col).drop(columns=[REP_Center])
        else:
            REP_Center= knut.get_unique_column_name("rep_center", right_input.schema) 
            right_gdf[REP_Center] = right_gdf.representative_point()
            right_gdf_temp = right_gdf.set_geometry(REP_Center).drop(
                columns=[self.right_geo_col]
            )
            gdf = gp.sjoin(left_gdf, right_gdf_temp, predicate="contains")